src/ui/network_game.rs

```diff
@@ -456,7 +459,8 @@ pub fn network_game_menu(
                                 .clicked()
                                 {
                                     *status = NetworkGameStatus::Searching;
-                                    online::start_search_for_game(matchmaking_server.clone(), *player_count);
+                                    let server = matchmaking_server.parse().expect("invalid server id");
```

Comment:
this should really not panic but show an error in the UI

Reply:
but I don't exactly know how, so I'll leave that up to you ;)

Reply:
yeah that's fine - we don't have great usage of friendly error display yet to model around.